src/dependency-manager/manager.ts

```diff
@@ -252,33 +252,33 @@ export default abstract class DependencyManager {
     }
   }
 
-  getSecretsForComponentSpec(component_spec: ComponentSpec, all_values: Dictionary<Dictionary<string | number | null>>): Dictionary<ParameterValue> {
+  getSecretsForComponentSpec(component_spec: ComponentSpec, all_secrets: Dictionary<Dictionary<string | number | null>>): Dictionary<SecretValue> {
     // pre-sort values dictionary to properly stack/override any colliding keys
-    const sorted_values_keys = Object.keys(all_values).sort();
+    const sorted_values_keys = Object.keys(all_secrets).sort();
     const sorted_values_dict: Dictionary<Dictionary<string | number | null>> = {};
     for (const key of sorted_values_keys) {
-      sorted_values_dict[key] = all_values[key];
+      sorted_values_dict[key] = all_secrets[key];
     }
 
     const component_ref = component_spec.metadata.ref;
     const { component_account_name, component_name, instance_name } = ComponentSlugUtils.parse(component_ref);
     const component_ref_with_account = component_account_name ? component_ref : ComponentSlugUtils.build(this.account, component_name, instance_name);
 
-    const component_parameters = new Set(Object.keys(component_spec.parameters || {}));
+    const component_secrets = new Set(Object.keys({ ...(component_spec.parameters || {}), ...(component_spec.secrets || {}) })); // TODO: 404: update
```

Comment:
I know you marked some of the other TODO's as resolved, but just wanted to makrk this one to be sure

Reply:
Thanks, yes this one was intentionally left there